### PR TITLE
HPCC-17970 Reschedule/Deschedule buttons enabling

### PR DESCRIPTION
**Pull request by miguelvazq**
The functionality for WUQuery list has changed so no longer do we support rescheduling/disabling from Workunit Query list. (https://github.com/hpcc-systems/HPCC-Platform/commit/50f58cd07a93fa3f6415b689c696ec216f9d1426) This change has to be done within the workunit details area and/or event schedule area (descheduling). I have modified the function call that we call to verify if there is an option to reschedule or deschedule.

Signed-off by: Miguel Vazquez <miguel.vazquez@lexisnexis.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [ ] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [x] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [x] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by miguelvazq** [comment]
@GordonSmith Please review. FYI @wangkx

**Review by GordonSmith — APPROVED** [review]
Looks good.

File: esp/src/eclwatch/WUDetailsWidget.js [hunk 1]
```diff
@@ -569,8 +571,8 @@ define([
             registry.byId(this.id + "Recover").set("disabled", isArchived || !this.wu.isComplete() || this.wu.isDeleted());
             registry.byId(this.id + "Publish").set("disabled", isArchived || !this.wu.isComplete() || this.wu.isDeleted());
             registry.byId(this.id + "ZapReport").set("disabled", this.wu.isDeleted());
-            registry.byId(this.id + "Reschedule").set("disabled", !this.wu.isComplete() || this.wu.isFailed());
-            registry.byId(this.id + "Deschedule").set("disabled", this.wu.isComplete() || this.wu.isFailed());
+            registry.byId(this.id + "Reschedule").set("disabled", !this.wu.isAbleToDeschedule());
```
Comment (richardkchapman) [inline esp/src/eclwatch/WUDetailsWidget.js]: Shouldn't this be calling isAbleToReschedule (and the one below it be calling isAbleToDeschedule) ?

**Comment by richardkchapman** [comment]
@miguelvazq One question

File: esp/src/eclwatch/WUDetailsWidget.js (hunk 1 above)
Comment (miguelvazq) [inline esp/src/eclwatch/WUDetailsWidget.js]: I am using the opposite of what each should have enabled.

Reschedule ---> you can de-schedule
Deschedule ---> you can re-schedule

Comment (richardkchapman) [inline esp/src/eclwatch/WUDetailsWidget.js]: If it's calling a function called "isAbleToDeschedule" then surely it's the deschedule button that should be enabled/disabled accordingly. Either the logic is wrong or the function name is confusing!

**Comment by miguelvazq** [comment]
@richardkchapman Updated.

**Pull request merged**